### DISCUSSION TITLE
Semantic versioning

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#394] [General]
[cb-operator](https://github.com/cloud-barista/cb-operator) repo 에서 테스트를 해 보았습니다.

[공유하고 싶은 사항]
- Pull requests 는 git 고유 기능이 아니라, git 고유 기능인 git merge 를 위한 **GitHub 기능**임
- 위와 비슷하게, Releases 는 git 고유 기능이 아니라, git 고유 기능인 git tag 를 위한 **GitHub 기능**임

[[Releases](https://github.com/cloud-barista/cb-operator/releases) and [Tags](https://github.com/cloud-barista/cb-operator/tags)]

|   | Release title | Tag version | 비고  |
|---|---|---|---|
| 2021/2/8  | v0.3.2-test (release for semver testing)  | v0.3.2-test  |   |
| 2021/2/4  | v0.3.1 (release for semver testing)  | v0.3.1  | Pre-release  |
| 2020/12/10  | Release for Cloud-Barista Espresso  | v0.3.0-espresso  |   |
| 2020/12/9  | cb-operator v0.2.9 (for staging espresso release)  | v0.2.9  | Pre-release  |
|   |   |   |   |

---

https://pkg.go.dev/github.com/cloud-barista/cb-operator
![image](https://user-images.githubusercontent.com/46767780/107171298-cc4e5b00-6a05-11eb-8ad4-97a51520af30.png)

---

`go help get`

```
...
For each named package or package pattern, get must decide which version of
the corresponding module to use. By default, get looks up the latest tagged
release version, such as v0.4.5 or v1.2.3. If there are no tagged release
versions, get looks up the latest tagged pre-release version, such as
v0.0.1-pre1. If there are no tagged versions at all, get looks up the latest
known commit. If the module is not already required at a later version
(for example, a pre-release newer than the latest release), get will use
the version it looked up. Otherwise, get will use the currently
required version.

```

[우선순위]
1. the latest tagged release version (예: v0.4.5 or v1.2.3)
2. (1번이 없으면) the latest tagged pre-release version (예: v0.0.1-pre1)
3. (1번도 2번도 없으면) the latest known commit

---

cb-operator 를 이용한 실험 및
CB-Spider, CB-Tumblebug 등의 tagging 현황을 보면

`go get`, `pkg.go.dev` 등은
`v0.3.0-espresso` 를 pre-release version 이라고 간주하여
`v0.2.9` 의 우선순위가 더 높다고 판단한 것입니다.

이는 Cloud-Barista 커뮤니티의 의도와는 다르다고 할 수 있겠습니다.

**Comment by seokho-son** [comment]
역시 그렇군요. 커뮤니티 차원에서 이슈화해보도록 하겠습니다. 검증 작업 감사합니다..^^

**Comment by seokho-son** [comment]
차기 릴리스부터는 시멘틱 버저닝을 준용하여,
v0.4.0 와 같은 형태로, 릴리스 tag의 명칭을 변경 진행하는 것으로 확정하였습니다. (CB 커뮤니티 차원)
제안 및 검토 감사합니다 ~ ! ^^

**Comment by yunkon-kim** [comment]
@jihoon-seo 님

테스트 및 공유해주신 컨텐츠가 좋아서 cb-coffeehouse([참고](https://github.com/cloud-barista/cb-coffeehouse/issues/2))에서 추후에 재 구성하여 공유하려고 합니다 😄

**Comment by jihoon-seo** [comment]
[참고] [유의적 버전 2.0.0-ko2](https://semver.org/lang/ko/)
#### “v1.2.3”은 유의적 버전 인가?
“v1.2.3”은 유의적 버전이 아니다. 그러나, 버전 숫자를 나타내기 위해 유의적 버전의 접두어로 “v”를 붙이기도 한다. 버전 관리 도구에서 “version”의 축약어로 “v”로 사용하는 것을 흔하게 볼 수 있다. 예: git tag v1.2.3 -m "Release version 1.2.3" 에서 “v1.2.3”은 tag name이고 유의적 버전은 “1.2.3” 이다.
